src/js/components/Header/Header.js

```diff
@@ -1,19 +1,59 @@
-import React from 'react';
-
+import React, { useContext, useEffect } from 'react';
+import { ThemeContext } from 'styled-components';
+import { findScrollParent, useForwardedRef } from '../../utils';
 import { Box } from '../Box';
 
-const Header = React.forwardRef(({ ...rest }, ref) => (
-  <Box
-    align="center"
-    as="header"
-    direction="row"
-    flex={false}
-    justify="between"
-    gap="medium"
-    {...rest}
-    ref={ref}
-  />
-));
+const Header = React.forwardRef(({ sticky, ...rest }, ref) => {
+  const theme = useContext(ThemeContext);
+  const headerRef = useForwardedRef(ref);
+  useEffect(() => {
+    let scrollTop = 0;
+    const updateScrollDir = () => {
+      // target is its scroll parent
+      const target = findScrollParent(headerRef.current);
+      const header = headerRef.current;
+
+      if (target && sticky === 'scrollup') {
+        const nextScrollTop =
+          target === document ? window.pageYOffset : target.scrollTop;
+        if (scrollTop - nextScrollTop <= 0) {
+          header.style.top = `-${header.getBoundingClientRect().height}px`;
+          header.style.zIndex = '';
+        } else if (scrollTop - nextScrollTop > 0) {
+          // ensure that header moves with the target
+          header.style.position = 'sticky';
+          header.style.top = '0px';
+          header.style.zIndex = `${theme.header?.sticky?.zIndex}`;
+          header.style.transition = 'top 0.6s';
+        }
+        scrollTop = nextScrollTop;
+      }
+    };
+
+    updateScrollDir();
+    window.addEventListener('resize', updateScrollDir);
```

Comment:
Can we wrap these addEventListeners() inside a test for `sticky === 'scrollup'`? I think that would make our code a bit more efficient.

Reply:
good thoughts doesn't need to run otherwise